lib/src/async-compiler.ts

```diff
@@ -6,42 +6,174 @@ import {spawn} from 'child_process';
 import {Observable} from 'rxjs';
 import {takeUntil} from 'rxjs/operators';
 
+import {
+  OptionsWithLegacy,
+  StringOptionsWithLegacy,
+  createDispatcher,
+  handleCompileResponse,
+  handleLogEvent,
+  newCompilePathRequest,
+  newCompileStringRequest,
+} from './compiler';
 import {compilerCommand} from './compiler-path';
+import {FunctionRegistry} from './function-registry';
+import {ImporterRegistry} from './importer-registry';
+import * as utils from './utils';
+import * as proto from './vendor/embedded_sass_pb';
+import {CompileResult} from './vendor/sass';
+import {PacketTransformer} from './packet-transformer';
+import {MessageTransformer} from './message-transformer';
 
 /**
- * An asynchronous wrapper for the embedded Sass compiler that exposes its stdio
- * streams as Observables.
+ * Flag allowing the constructor passed by `initAsyncCompiler` so we can
+ * differentiate and throw an error if the `AsyncCompiler` is constructed via
+ * `new AsyncCompiler`.
  */
-export class AsyncEmbeddedCompiler {
+const initFlag = Symbol();
+
+/**
+ * An asynchronous wrapper for the embedded Sass compiler
+ */
+export class AsyncCompiler {
   /** The underlying process that's being wrapped. */
   private readonly process = spawn(
     compilerCommand[0],
     [...compilerCommand.slice(1), '--embedded'],
     {windowsHide: true}
   );
 
+  /** The next compilation ID */
+  private compilationId = 1;
+
+  /** A list of active compilations */
+  private compilations: Set<Promise<proto.OutboundMessage_CompileResponse>> =
```

Comment:
This is never reassigned, so
```suggestion
private readonly compilations: Set<Promise<proto.OutboundMessage_CompileResponse>> =
```

Reply:
Fixed in bf7cf3c68097e93ce73693a2fa418870ae4aa65b